### PR TITLE
Add support for puppetlabs/puppetlabs-strings

### LINKED ISSUE #33: Add support for puppetlabs/puppet-strings
This module should contain a ```REFERENCE.md```.

### PR DESCRIPTION
**Pull request by dhoppe**
Fixes #33, #44

### CODE REVIEW

File: .sync.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@
   docker_sets:
     - set: centos6-64
     - set: centos7-64
+    - set: centos8-64
```
Comment (ekohl) [inline .sync.yml:6]: This looks unrelated to strings

File: README.md [hunk 2]
```diff
@@ -8,84 +9,98 @@
...
 
-## Usage
+* Caddy binary
+* Caddy configuration file
+* Caddy virtual hosts
+* Caddy service
+
+### Setup Requirements
+
+This module has the following dependencies:
```
Comment (ekohl) [inline README.md:39]: Do we need to list these? IMHO metadata.json is a better place.

File: manifests/init.pp [hunk 3]
```diff
@@ -1,53 +1,84 @@
...
-  Stdlib::Absolutepath           $caddy_log_dir = '/var/log/caddy',
-  Stdlib::Absolutepath           $caddy_tmp_dir = '/tmp',
-  Stdlib::Absolutepath           $caddy_home = '/etc/ssl/caddy',
-  Stdlib::Absolutepath           $caddy_ssl_dir = "${caddy_home}/.caddy",
-  Enum['personal', 'commercial'] $caddy_license = 'personal',
-  Enum['on','off']               $caddy_telemetry = 'off',
-  String                         $caddy_features = 'http.filter,http.git,http.ipfilter',
-  Stdlib::Port                   $caddy_http_port = 80,
-  Stdlib::Port                   $caddy_https_port = 443,
+  Stdlib::Absolutepath           $install_path          = '/usr/local/bin',
```
Comment (ekohl) [inline manifests/init.pp:64]: I have stopped aligning because I think it just makes things hard to maintain. Won't object, but I wouldn't bother.

File: manifests/service.pp [hunk 4]
```diff
@@ -26,8 +26,8 @@
       }
       ~> Service['caddy']
     }
-    'redhat': { # we could probably add 'debian' for older debian releases but not sure
-      file {'/etc/init.d/caddy':
+    'redhat': {
+      file { '/etc/init.d/caddy':
         ensure  => file,
         mode    => '0744',
```
Comment (ekohl) [inline manifests/service.pp:32]: Unrelated, but what a weird mode. `0755` is much more logical IMHO.

File: manifests/vhost.pp [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
 #
+# @example Configure virtual host, based on content
+#   caddy::vhost { 'example2:
+#     content => 'localhost:2015',
+#   }
 #
 define caddy::vhost(
-  $source     = undef,
-  $content    = undef,
+  $source  = undef,
```
Comment (ekohl) [inline manifests/vhost.pp:15]: Undocumented, also
```suggestion
  Optional[Stdlib::Filesource] $source = undef,
```

File: manifests/vhost.pp [hunk 6]
```diff
@@ -1,19 +1,19 @@
...
+  $content = undef,
```
Comment (ekohl) [inline manifests/vhost.pp:16]: Undocumented, also
```suggestion
  Optional[String] $content = undef,
```

File: .sync.yml (hunk 1 above)
Comment (dhoppe) [reply .sync.yml:6]: That was missing at a pull request I merged earlier today.

File: README.md (hunk 2 above)
Comment (dhoppe) [reply README.md:39]: I have no idea. We have these dependencies at ```metadata.json```, but it is nice to get an quick overview without jumping through all these files. 

However, it is almost impossible to create a good ```README.md```, because all of our modules use a different style and we do not have a reference module like ```puppetlabs/puppetlabs-apt```.

File: manifests/init.pp (hunk 3 above)
Comment (dhoppe) [reply manifests/init.pp:64]: Yeah, this mess came with the data types, but I think it is easier to read this way.

File: manifests/service.pp (hunk 4 above)
Comment (dhoppe) [reply manifests/service.pp:32]: Yeah, this was already in here and nobody cared about it. I will change it. ;)

File: manifests/vhost.pp (hunk 5 above)
Comment (dhoppe) [reply manifests/vhost.pp:15]: Thx. I missed that somehow.

File: manifests/vhost.pp (hunk 6 above)
Comment (dhoppe) [reply manifests/vhost.pp:16]: Thx. I missed that somehow.

**Comment by dhoppe** [comment]
@ekohl Just generated a new ```REFERENCE.md```.

**Comment by ghoneycutt** [comment]
Seems to duplicate #49 which was merged. Please re-open if that's not the case.
